### PR TITLE
fix: change namespace starts processor on namespace change even if not leader

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

**Comment by csviri** [comment]
Would be a bit nicer to check if the instance is the leader, but this should solve the problem for now. Will add a test for this use case.

**Review by metacosm — APPROVED** [review]
LGTM

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java [hunk 1]
```diff
@@ -368,17 +368,25 @@ private void validateCRDWithLocalModelIfRequired(Class<P> resClass, String contr
...
+    // mainly start the processor on namespace change.
+    boolean eventProcessorWasRunning = eventProcessor.isRunning();
+    if (eventProcessorWasRunning) {
+      eventProcessor.stop();
+    }
     eventSourceManager.changeNamespaces(namespaces);
-    eventProcessor.start();
+    if (eventProcessorWasRunning) {
+      eventProcessor.start();
+    }
```
Comment (gyfora) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java:389]: could it happen that at this point we are are already the leader (and accidentally not start processing) or the `synchronised` will take care of that

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java:389]: exactly, `synchronized` takes care of that

**Review by gyfora — APPROVED** [review]

**Pull request merged**